tools/filterwriter/filterwriter.go

```diff
@@ -20,17 +20,19 @@ type Writer struct {
 	writer  io.Writer
 	secrets [][][]byte
 
-	chunk []byte
-	store [][]byte
-	mux   sync.Mutex
+	chunk     []byte
+	store     [][]byte
+	mux       sync.Mutex
+	lastTimer *time.Timer
 }
 
 // New ...
 func New(secrets []string, target io.Writer) *Writer {
-	return &Writer{
+	writer := &Writer{
 		writer:  target,
 		secrets: secretsByteList(secrets),
 	}
+	return writer
```

Comment:
Is there a reason for creating a writer and return it instead of return the new writer object immediately?

Reply:
very good catch, it was previously but I forgot to convert it back, thanks!